src/control/server/engine/config.go

```diff
@@ -170,42 +169,43 @@ func NewConfig() *Config {
 	}
 }
 
-// ValidateFabric validates the fabric configuration against actual fabric devices and NUMA config.
-func (c *Config) ValidateFabric(ctx context.Context, log logging.Logger, fis *hardware.FabricInterfaceSet) error {
-	fi, err := fis.GetInterfaceOnOSDevice(c.Fabric.Interface, c.Fabric.Provider)
-	if err != nil {
-		return err
-	}
-
-	return errors.Wrap(c.setAffinity(ctx, log, fi), "setting numa affinity for engine")
-}
-
 // setAffinity ensures engine NUMA locality is assigned and valid.
-func (c *Config) setAffinity(ctx context.Context, log logging.Logger, fi *hardware.FabricInterface) error {
-	ifaceNumaNode := fi.NUMANode
+func (c *Config) setAffinity(log logging.Logger, fis *hardware.FabricInterfaceSet) (err error) {
+	var fi *hardware.FabricInterface
+	if fis != nil {
```

Comment:
Is fis == nil a valid use case?

Reply:
Sometimes we call validate without having access to the FabricInterfaceSet, i.e. when we are calling from lib/control through dmg on a client rather than a storage server. We could remedy this situation by providing a mechanism for remote network scan to provide a FIS (or helper to translate the results back into one) but I thought that it was better to allow setAffinity to use FIS if provided but not require it.